src/libcrun/container.c

```diff
@@ -953,6 +956,102 @@ wasmer_do_exec (void *container, void *arg, const char *pathname, char *const ar
+
+  WasmEdge_ConfigureCreate = dlsym (handle, "WasmEdge_ConfigureCreate");
+  WasmEdge_ConfigureDelete = dlsym (handle, "WasmEdge_ConfigureDelete");
+  WasmEdge_ConfigureAddProposal = dlsym (handle, "WasmEdge_ConfigureAddProposal");
+  WasmEdge_ConfigureAddHostRegistration = dlsym (handle, "WasmEdge_ConfigureAddHostRegistration");
+  WasmEdge_VMCreate = dlsym (handle, "WasmEdge_VMCreate");
+  WasmEdge_VMDelete = dlsym (handle, "WasmEdge_VMDelete");
+  WasmEdge_VMRegisterModuleFromFile = dlsym (handle, "WasmEdge_VMRegisterModuleFromFile");
+  WasmEdge_VMGetImportModuleContext = dlsym (handle, "WasmEdge_VMGetImportModuleContext");
+  WasmEdge_ImportObjectInitWASI = dlsym (handle, "WasmEdge_ImportObjectInitWASI");
+  WasmEdge_ImportObjectInitWasmEdgeProcess = dlsym (handle, "WasmEdge_ImportObjectInitWasmEdgeProcess");
+  WasmEdge_VMRunWasmFromFile = dlsym (handle, "WasmEdge_VMRunWasmFromFile");
+  WasmEdge_ResultOK = dlsym (handle, "WasmEdge_ResultOK");
+  WasmEdge_StringCreateByCString = dlsym (handle, "WasmEdge_StringCreateByCString");
+  if (WasmEdge_ConfigureCreate == NULL || WasmEdge_ConfigureDelete == NULL || WasmEdge_ConfigureAddProposal == NULL || WasmEdge_ConfigureAddHostRegistration == NULL || WasmEdge_VMCreate == NULL || WasmEdge_VMDelete == NULL || WasmEdge_VMRegisterModuleFromFile == NULL || WasmEdge_VMGetImportModuleContext == NULL || WasmEdge_ImportObjectInitWASI == NULL || WasmEdge_ImportObjectInitWasmEdgeProcess == NULL || WasmEdge_VMRunWasmFromFile == NULL || WasmEdge_ResultOK == NULL || WasmEdge_StringCreateByCString == NULL)
+    {
+      fprintf (stderr, "could not find symbol in `libwasmedge.so`");
+      dlclose (handle);
+      return -1;
+    }
+
+  WasmEdge_ConfigureContext *configure = WasmEdge_ConfigureCreate ();
+  if (UNLIKELY (configure == NULL))
+    error (EXIT_FAILURE, 0, "could not create wasmedge configure");
+
+  WasmEdge_ConfigureAddProposal (configure, WasmEdge_Proposal_BulkMemoryOperations);
+  WasmEdge_ConfigureAddProposal (configure, WasmEdge_Proposal_ReferenceTypes);
+  WasmEdge_ConfigureAddProposal (configure, WasmEdge_Proposal_SIMD);
+  WasmEdge_ConfigureAddHostRegistration (configure, WasmEdge_HostRegistration_Wasi);
+  WasmEdge_ConfigureAddHostRegistration (configure, WasmEdge_HostRegistration_WasmEdge_Process);
+
+  WasmEdge_VMContext *vm = WasmEdge_VMCreate (configure, NULL);
+  if (UNLIKELY (vm == NULL))
+    {
+      WasmEdge_ConfigureDelete (configure);
+      error (EXIT_FAILURE, 0, "could not create wasmedge vm");
+    }
+
+  WasmEdge_ImportObjectContext *wasi_module = WasmEdge_VMGetImportModuleContext (vm, WasmEdge_HostRegistration_Wasi);
+  if (UNLIKELY (wasi_module == NULL))
+    {
+      WasmEdge_VMDelete (vm);
+      WasmEdge_ConfigureDelete (configure);
+      error (EXIT_FAILURE, 0, "could not get wasmedge wasi module context");
+    }
+
+  WasmEdge_ImportObjectContext *proc_module = WasmEdge_VMGetImportModuleContext (vm, WasmEdge_HostRegistration_WasmEdge_Process);
```

Comment:
nit: we usually define variables at the beginning of the scope (same comment applies to other lines), but I am fine to relax this requirement or fix it as a follow up

Reply:
Done.